source/draw/RasterMonochrome.ooc

```diff
@@ -222,6 +222,36 @@ RasterMonochrome: class extends RasterPacked {
 		}
 		result
 	}
+	getCrop: func (region: IntBox2D) -> This {
+		result := This new(region size)
+		destination := result buffer pointer
+		source := this buffer pointer + region top * this stride + region left
+		for (y in 0 .. region height) {
+			for (x in 0 .. region width) {
+				destination@ = source@
+				destination += 1
+				source += 1
+			}
+			destination += result stride - result width
+			source += this stride - region width
+		}
+		result
+	}
+	averagePixelDifference: func (other: This, padding := 0) -> Float {
+		difference := 0.0f
+		thisBuffer := this buffer pointer + this stride * padding + padding
+		otherBuffer := other buffer pointer + other stride * padding + padding
+		for (y in padding .. this height - padding) {
+			for (x in padding .. this width - padding) {
```

Comment:
Since you seem to care about performance, you should also precompute `this height - padding` and `this width - padding` into local variables.